src/vcs.rs

```diff
@@ -1,8 +1,107 @@
+use std::env;
+
+use anyhow::bail;
 use clap::Parser;
 
+use crate::CargoResult;
+
 #[derive(Parser, Debug)]
-pub struct VCSopts {
+pub struct VcsOpts {
     /// Fix code even if a VCS was not detected
     #[arg(long)]
     pub allow_no_vcs: bool,
+
+    /// Fix code even if the working directory is dirty or has staged changes
+    #[arg(long)]
+    pub allow_dirty: bool,
+
+    /// Fix code even if the working directory has staged changes
+    #[arg(long)]
+    pub allow_staged: bool,
+}
+
+impl VcsOpts {
+    pub fn valid_vcs(&self) -> CargoResult<()> {
+        if self.allow_no_vcs {
+            return Ok(());
+        }
+        let cwd = env::current_dir()?;
+
+        let repo = git2::Repository::discover_path(&cwd, &cwd)
+            .ok()
+            .and(git2::Repository::discover(&cwd).ok())
+            .filter(|r| {
+                if r.workdir().is_some_and(|workdir| workdir == cwd) {
+                    true
+                } else {
+                    !r.is_path_ignored(cwd).unwrap_or(false)
+                }
+            });
+
+        if let Some(repo) = repo {
```

Comment:
```rust
let Some(repo) = repo else {
    // ...
};
```
Generally you don't want to indent your code for bookkeeping like this.  Reserving it only for important distinctions helps draw the eye to it more, making it more scannable